### ISSUE TITLE
Specify the endpoint of existed Spark master

### ISSUE THREAD

**Issue by JiaMingLin** [datalayer/zeppelin-docker#1]
Hi
I have a Spark cluster already, and I'd like to use your Zeppelin in Docker to work with my Spark cluster
, but how can I specify the endpoint of my Spark Master?

**Comment by echarles** [comment]
I have pushed updates to allow configuration via environment variables:
DOCKER_SPARK_MASTER = MASTER for Spark (default is local[*]).
DOCKER_HADOOP_CONF_DIR = The folder for the Hadoop configuration file (default is /etc/hadoop/conf).

Example: DOCKER_ZEPPELIN_PORT=8081 ./zeppelin-docker-start

PS: If you need to change the Hadoop configuration, you will have to rebuild the Docker image

**Comment by JiaMingLin** [comment]
Thank you

**Comment by l15k4** [comment]
Actually it won't work because `local[*]` is declared here : 
https://github.com/datalayer/zeppelin-docker/blob/master/master/Dockerfile#L156

And it is preferred over any other settings. this `interpreter.json` shouldn't be part of the docker image imho.

**Comment by echarles** [comment]
@l15k4 makes sense. I have pushed `35375324ae07aaf7b83066f9f3107677bf730687` which doesn't add anymore the interpreter.json.

**Comment by oluies** [comment]
If I want to run against yarn, do I have to rebuild it with -Pyarn?

**Comment by echarles** [comment]
To run on Yarn:

Option 1: Change the hadoop configuration (in zeppelin/resources/hadoop), rebuild the docker image and launch with DOCKER_SPARK_MASTER=yarn-client

Option 2: Start the docker image, connect via SSH (ssh root@localhost -p 2222), stop zeppelin, define your hadoop configuration in a new folder, export HADOOP_CONF_DIR to your new configuration, export MASTER=yarn-client and start zeppelin.
